File: Easy/392. Is Subsequence/392. Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int sPtr = 0;
        int tPtr = 0;

        while (sPtr < s.length() && tPtr < t.length()) {
            if (s.charAt(sPtr) == t.charAt(tPtr)) {
                sPtr++;
            }
            tPtr++;
        }

        return sPtr == s.length();
    }
}

File: Easy/392. Is Subsequence/392. Is Subsequence.rs
impl Solution {
    pub fn is_subsequence(s: String, t: String) -> bool {
        let s_chars: Vec<char> = s.chars().collect();
        let t_chars: Vec<char> = t.chars().collect();
        let mut s_ptr = 0;
        let mut t_ptr = 0;

        while s_ptr < s_chars.len() && t_ptr < t_chars.len() {
            if s_chars[s_ptr] == t_chars[t_ptr] {
                s_ptr += 1;
            }
            t_ptr += 1;
        }

        s_ptr == s_chars.len()
    }
}
